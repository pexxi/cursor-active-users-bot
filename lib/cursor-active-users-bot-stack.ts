import * as path from "node:path";
import * as cdk from "aws-cdk-lib";
import { SecretValue } from "aws-cdk-lib";
import * as events from "aws-cdk-lib/aws-events";
import * as targets from "aws-cdk-lib/aws-events-targets";
import * as lambda from "aws-cdk-lib/aws-lambda";
import { NodejsFunction } from "aws-cdk-lib/aws-lambda-nodejs";
import * as secretsmanager from "aws-cdk-lib/aws-secretsmanager";
import type { Construct } from "constructs";

interface CursorActiveUsersBotStackProps extends cdk.StackProps {
	env?: {
		account: string;
		region: string;
	};
}

export class CursorActiveUsersBotStack extends cdk.Stack {
	constructor(
		scope: Construct,
		id: string,
		props?: CursorActiveUsersBotStackProps,
	) {
		super(scope, id, props);

		// For testing, we can get account and region from props.env
		// For actual deployment, we need them from environment variables
		const account = props?.env?.account || process.env.AWS_ACCOUNT;
		const region = props?.env?.region || process.env.AWS_REGION;

		// Only check environment variables if not provided via props (for actual deployment)
		if (!props?.env && (!process.env.AWS_ACCOUNT || !process.env.AWS_REGION)) {
			throw new Error(
				"Missing required environment variables in .env file for CDK stack. Please check AWS_ACCOUNT, and AWS_REGION.",
			);
		}

		// Create a new secret for API keys
		const apiSecrets = new secretsmanager.Secret(this, "ApiSecrets", {
			secretName: "CursorActiveUserBotSecrets",
			description: "API keys for Cursor and Slack bots",
			secretObjectValue: {
				CURSOR_API_KEY: SecretValue.unsafePlainText(
					"YOUR_CURSOR_API_KEY_PLACEHOLDER",
				),
				SLACK_BOT_TOKEN: SecretValue.unsafePlainText(
					"YOUR_SLACK_BOT_TOKEN_PLACEHOLDER",
				),
				SLACK_USER_ID: SecretValue.unsafePlainText(
					"YOUR_SLACK_USER_ID_PLACEHOLDER",
				),
				SLACK_SIGNING_SECRET: SecretValue.unsafePlainText(
					"YOUR_SLACK_SIGNING_SECRET_PLACEHOLDER",
				),
			},
		});

		// Define the Lambda function
		const inactiveUserCheckerLambda = new NodejsFunction(
			this,
			"InactiveUserCheckerFunction",
			{
				runtime: lambda.Runtime.NODEJS_22_X,
				entry: path.join(__dirname, "../src/lambda/index.ts"), // Path to the lambda handler file
				handler: "handler", // Function name in index.ts
				timeout: cdk.Duration.minutes(1), // Adjust timeout as needed
				memorySize: 256, // Adjust memory as needed
				environment: {
					SECRETS_ARN: apiSecrets.secretArn,
					INACTIVITY_MONTHS: process.env.INACTIVITY_MONTHS || "",
					// AWS_ACCOUNT and AWS_REGION are typically implicitly handled by CDK based on context/profile
					// but can be passed if the lambda needs them for other AWS SDK calls unrelated to its own execution region/account.
				},
			},
		);

		// Grant the Lambda function read access to the secret
		apiSecrets.grantRead(inactiveUserCheckerLambda);

		// Define the EventBridge rule to run the Lambda monthly
		// Runs at 00:00 UTC on the 1st day of every month
		const rule = new events.Rule(this, "MonthlyInactiveUserCheckRule", {
			schedule: events.Schedule.cron({
				minute: "0",
				hour: "0",
				day: "1",
				month: "*", // Every month
				year: "*", // Every year
			}),
			description:
				"Triggers the inactive user checker Lambda function monthly.",
		});

		// Set the Lambda function as the target for the EventBridge rule
		rule.addTarget(new targets.LambdaFunction(inactiveUserCheckerLambda));

		// Output the Lambda function name
		new cdk.CfnOutput(this, "LambdaFunctionName", {
			value: inactiveUserCheckerLambda.functionName,
			description: "Name of the Inactive User Checker Lambda function",
		});
	}
}
