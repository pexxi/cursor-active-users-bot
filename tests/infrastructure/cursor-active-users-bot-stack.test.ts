import * as cdk from "aws-cdk-lib";
import { Template } from "aws-cdk-lib/assertions";
import { CursorActiveUsersBotStack } from "../../lib/cursor-active-users-bot-stack";
import { processS3Keys } from "../serializers/s3-key-serializer";

// Mock environment variables for consistent testing
const mockEnv = {
	AWS_ACCOUNT: "123456789012",
	AWS_REGION: "us-east-1",
};

describe("CursorActiveUsersBotStack Infrastructure Tests", () => {
	let app: cdk.App;
	let stack: CursorActiveUsersBotStack;
	let template: Template;

	beforeEach(() => {
		// Create CDK app and stack with explicit environment configuration
		// NOTE: We pass environment values directly to the stack instead of mutating process.env
		app = new cdk.App();
		stack = new CursorActiveUsersBotStack(app, "TestStack", {
			env: {
				account: mockEnv.AWS_ACCOUNT,
				region: mockEnv.AWS_REGION,
			},
		});
		template = Template.fromStack(stack);
	});

	describe("Complete Stack Snapshot", () => {
		it("should match the complete CloudFormation template snapshot", () => {
			const templateJson = template.toJSON();
			expect(processS3Keys(templateJson)).toMatchSnapshot();
		});
	});

	describe("Lambda Function", () => {
		it("should create a Lambda function with correct properties", () => {
			template.hasResourceProperties("AWS::Lambda::Function", {
				Runtime: "nodejs22.x",
				Handler: "index.handler",
				MemorySize: 512,
				Timeout: 300,
			});
		});

		it("should have environment variables for configuration", () => {
			const lambdaResources = template.findResources("AWS::Lambda::Function");
			const lambdaResource = Object.values(lambdaResources)[0] as any;

			expect(
				lambdaResource.Properties.Environment.Variables.SECRETS_ARN,
			).toBeDefined();
			expect(
				lambdaResource.Properties.Environment.Variables.SECRETS_ARN.Ref,
			).toBeDefined();
			expect(
				lambdaResource.Properties.Environment.Variables.NOTIFY_AFTER_DAYS,
			).toBe("60");
			expect(
				lambdaResource.Properties.Environment.Variables.REMOVE_AFTER_DAYS,
			).toBe("90");
		});

		it("should match Lambda function snapshot", () => {
			const lambdaResources = template.findResources("AWS::Lambda::Function");
			expect(processS3Keys(lambdaResources)).toMatchSnapshot(
				"lambda-functions",
			);
		});
	});

	describe("Secrets Manager", () => {
		it("should create a secret with correct name and description", () => {
			template.hasResourceProperties("AWS::SecretsManager::Secret", {
				Name: "CursorActiveUserBotSecrets",
				Description: "API keys for Cursor and Slack bots",
			});
		});

		it("should create secret with placeholder values", () => {
			// Verify the secret contains the expected keys
			const secretResources = template.findResources(
				"AWS::SecretsManager::Secret",
			);
			const secretResource = Object.values(secretResources)[0] as any;
			const secretString = secretResource.Properties.SecretString;

			expect(secretString).toBeDefined();
			expect(secretString).toContain("CURSOR_API_KEY");
			expect(secretString).toContain("SLACK_BOT_TOKEN");
			expect(secretString).toContain("SLACK_USER_ID");
		});

		it("should match Secrets Manager snapshot", () => {
			const secretResources = template.findResources(
				"AWS::SecretsManager::Secret",
			);
			expect(processS3Keys(secretResources)).toMatchSnapshot("secrets-manager");
		});
	});

	describe("EventBridge Rule", () => {
		it("should create a scheduled rule for weekly execution", () => {
			template.hasResourceProperties("AWS::Events::Rule", {
				ScheduleExpression: "cron(0 9 ? * MON *)",
				Description:
					"Triggers the inactive user checker Lambda function weekly on Mondays at 9 AM UTC.",
				State: "ENABLED",
			});
		});

		it("should target the Lambda function", () => {
			const ruleResources = template.findResources("AWS::Events::Rule");
			const ruleResource = Object.values(ruleResources)[0] as any;

			expect(ruleResource.Properties.Targets).toHaveLength(1);
			expect(
				ruleResource.Properties.Targets[0].Arn["Fn::GetAtt"],
			).toBeDefined();
			expect(ruleResource.Properties.Targets[0].Arn["Fn::GetAtt"][1]).toBe(
				"Arn",
			);
			expect(ruleResource.Properties.Targets[0].Id).toBe("Target0");
		});

		it("should match EventBridge rule snapshot", () => {
			const ruleResources = template.findResources("AWS::Events::Rule");
			expect(processS3Keys(ruleResources)).toMatchSnapshot("eventbridge-rules");
		});
	});

	describe("IAM Permissions", () => {
		it("should create IAM role for Lambda execution", () => {
			template.hasResourceProperties("AWS::IAM::Role", {
				AssumeRolePolicyDocument: {
					Statement: [
						{
							Action: "sts:AssumeRole",
							Effect: "Allow",
							Principal: {
								Service: "lambda.amazonaws.com",
							},
						},
					],
				},
			});
		});

		it("should grant Lambda permission to read secrets", () => {
			const policyResources = template.findResources("AWS::IAM::Policy");
			const policyResource = Object.values(policyResources)[0] as any;

			const statements = policyResource.Properties.PolicyDocument.Statement;
			const secretsStatement = statements.find((stmt: { Action?: string }) =>
				stmt.Action?.includes("secretsmanager:GetSecretValue"),
			);

			expect(secretsStatement).toBeDefined();
			expect(secretsStatement.Effect).toBe("Allow");
		});

		it("should create Lambda permission for EventBridge", () => {
			template.hasResourceProperties("AWS::Lambda::Permission", {
				Action: "lambda:InvokeFunction",
				Principal: "events.amazonaws.com",
			});
		});

		it("should match IAM resources snapshot", () => {
			const iamRoles = template.findResources("AWS::IAM::Role");
			const iamPolicies = template.findResources("AWS::IAM::Policy");
			const lambdaPermissions = template.findResources(
				"AWS::Lambda::Permission",
			);

			expect(
				processS3Keys({
					roles: iamRoles,
					policies: iamPolicies,
					permissions: lambdaPermissions,
				}),
			).toMatchSnapshot("iam-resources");
		});
	});

	describe("CloudFormation Outputs", () => {
		it("should create output for Lambda function name", () => {
			template.hasOutput("LambdaFunctionName", {
				Description: "Name of the Inactive User Checker Lambda function",
			});
		});

		it("should create output for schedule information", () => {
			template.hasOutput("ScheduleInfo", {
				Description: "Lambda execution schedule",
			});
		});

		it("should match outputs snapshot", () => {
			const outputs = template.toJSON().Outputs;
			expect(processS3Keys(outputs)).toMatchSnapshot("cloudformation-outputs");
		});
	});

	describe("Resource Counts", () => {
		it("should create the expected number of resources", () => {
			const resources = template.toJSON().Resources;
			const resourceCounts = Object.keys(resources).reduce(
				(acc, key) => {
					const resourceType = resources[key].Type;
					acc[resourceType] = (acc[resourceType] || 0) + 1;
					return acc;
				},
				{} as Record<string, number>,
			);

			expect(processS3Keys(resourceCounts)).toMatchSnapshot("resource-counts");
		});
	});

	describe("Stack Properties", () => {
		it("should have correct stack properties", () => {
			expect(stack.account).toBe(mockEnv.AWS_ACCOUNT);
			expect(stack.region).toBe(mockEnv.AWS_REGION);
			expect(stack.stackName).toBe("TestStack");
		});
	});
});
