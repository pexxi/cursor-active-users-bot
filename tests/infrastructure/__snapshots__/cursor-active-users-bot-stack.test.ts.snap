// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`CursorActiveUsersBotStack Infrastructure Tests CloudFormation Outputs should match outputs snapshot: cloudformation-outputs 1`] = `
{
  "LambdaFunctionName": {
    "Description": "Name of the Inactive User Checker Lambda function",
    "Value": {
      "Ref": "InactiveUserCheckerFunctionD51D6C82",
    },
  },
  "ScheduleInfo": {
    "Description": "Lambda execution schedule",
    "Value": "Runs weekly on Mondays at 9:00 AM UTC",
  },
}
`;

exports[`CursorActiveUsersBotStack Infrastructure Tests Complete Stack Snapshot should match the complete CloudFormation template snapshot 1`] = `
{
  "Outputs": {
    "LambdaFunctionName": {
      "Description": "Name of the Inactive User Checker Lambda function",
      "Value": {
        "Ref": "InactiveUserCheckerFunctionD51D6C82",
      },
    },
    "ScheduleInfo": {
      "Description": "Lambda execution schedule",
      "Value": "Runs weekly on Mondays at 9:00 AM UTC",
    },
  },
  "Parameters": {
    "BootstrapVersion": {
      "Default": "/cdk-bootstrap/hnb659fds/version",
      "Description": "Version of the CDK Bootstrap resources in this environment, automatically retrieved from SSM Parameter Store. [cdk:skip]",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
  },
  "Resources": {
    "ApiSecretsED109BDA": {
      "DeletionPolicy": "Delete",
      "Properties": {
        "Description": "API keys for Cursor and Slack bots",
        "Name": "CursorActiveUserBotSecrets",
        "SecretString": "{"CURSOR_API_KEY":"YOUR_CURSOR_API_KEY_PLACEHOLDER","GITHUB_TOKEN":"YOUR_GITHUB_TOKEN_PLACEHOLDER","GITHUB_ORG":"YOUR_GITHUB_ORG_PLACEHOLDER","SLACK_BOT_TOKEN":"YOUR_SLACK_BOT_TOKEN_PLACEHOLDER","SLACK_USER_ID":"YOUR_SLACK_USER_ID_PLACEHOLDER","SLACK_SIGNING_SECRET":"YOUR_SLACK_SIGNING_SECRET_PLACEHOLDER"}",
      },
      "Type": "AWS::SecretsManager::Secret",
      "UpdateReplacePolicy": "Delete",
    },
    "InactiveUserCheckerFunctionD51D6C82": {
      "DependsOn": [
        "InactiveUserCheckerFunctionServiceRoleDefaultPolicy3B2516D9",
        "InactiveUserCheckerFunctionServiceRoleF228211B",
      ],
      "Properties": {
        "Code": {
          "S3Bucket": "cdk-hnb659fds-assets-123456789012-us-east-1",
          "S3Key": "STATIC_ASSET_KEY.zip",
        },
        "Environment": {
          "Variables": {
            "ENABLE_CURSOR": "true",
            "ENABLE_GITHUB_COPILOT": "true",
            "ENABLE_NOTIFICATIONS": "true",
            "NOTIFY_AFTER_DAYS": "60",
            "REMOVE_AFTER_DAYS": "90",
            "SECRETS_ARN": {
              "Ref": "ApiSecretsED109BDA",
            },
          },
        },
        "Handler": "index.handler",
        "MemorySize": 512,
        "Role": {
          "Fn::GetAtt": [
            "InactiveUserCheckerFunctionServiceRoleF228211B",
            "Arn",
          ],
        },
        "Runtime": "nodejs22.x",
        "Timeout": 300,
      },
      "Type": "AWS::Lambda::Function",
    },
    "InactiveUserCheckerFunctionServiceRoleDefaultPolicy3B2516D9": {
      "Properties": {
        "PolicyDocument": {
          "Statement": [
            {
              "Action": [
                "secretsmanager:GetSecretValue",
                "secretsmanager:DescribeSecret",
              ],
              "Effect": "Allow",
              "Resource": {
                "Ref": "ApiSecretsED109BDA",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "InactiveUserCheckerFunctionServiceRoleDefaultPolicy3B2516D9",
        "Roles": [
          {
            "Ref": "InactiveUserCheckerFunctionServiceRoleF228211B",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "InactiveUserCheckerFunctionServiceRoleF228211B": {
      "Properties": {
        "AssumeRolePolicyDocument": {
          "Statement": [
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": [
          {
            "Fn::Join": [
              "",
              [
                "arn:",
                {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
              ],
            ],
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "WeeklyInactiveUserCheckRuleAllowEventRuleTestStackInactiveUserCheckerFunction4054795B2F7A268A": {
      "Properties": {
        "Action": "lambda:InvokeFunction",
        "FunctionName": {
          "Fn::GetAtt": [
            "InactiveUserCheckerFunctionD51D6C82",
            "Arn",
          ],
        },
        "Principal": "events.amazonaws.com",
        "SourceArn": {
          "Fn::GetAtt": [
            "WeeklyInactiveUserCheckRuleFB167B9C",
            "Arn",
          ],
        },
      },
      "Type": "AWS::Lambda::Permission",
    },
    "WeeklyInactiveUserCheckRuleFB167B9C": {
      "Properties": {
        "Description": "Triggers the inactive user checker Lambda function weekly on Mondays at 9 AM UTC.",
        "ScheduleExpression": "cron(0 9 ? * MON *)",
        "State": "ENABLED",
        "Targets": [
          {
            "Arn": {
              "Fn::GetAtt": [
                "InactiveUserCheckerFunctionD51D6C82",
                "Arn",
              ],
            },
            "Id": "Target0",
          },
        ],
      },
      "Type": "AWS::Events::Rule",
    },
  },
  "Rules": {
    "CheckBootstrapVersion": {
      "Assertions": [
        {
          "Assert": {
            "Fn::Not": [
              {
                "Fn::Contains": [
                  [
                    "1",
                    "2",
                    "3",
                    "4",
                    "5",
                  ],
                  {
                    "Ref": "BootstrapVersion",
                  },
                ],
              },
            ],
          },
          "AssertDescription": "CDK bootstrap stack version 6 required. Please run 'cdk bootstrap' with a recent version of the CDK CLI.",
        },
      ],
    },
  },
}
`;

exports[`CursorActiveUsersBotStack Infrastructure Tests EventBridge Rule should match EventBridge rule snapshot: eventbridge-rules 1`] = `
{
  "WeeklyInactiveUserCheckRuleFB167B9C": {
    "Properties": {
      "Description": "Triggers the inactive user checker Lambda function weekly on Mondays at 9 AM UTC.",
      "ScheduleExpression": "cron(0 9 ? * MON *)",
      "State": "ENABLED",
      "Targets": [
        {
          "Arn": {
            "Fn::GetAtt": [
              "InactiveUserCheckerFunctionD51D6C82",
              "Arn",
            ],
          },
          "Id": "Target0",
        },
      ],
    },
    "Type": "AWS::Events::Rule",
  },
}
`;

exports[`CursorActiveUsersBotStack Infrastructure Tests IAM Permissions should match IAM resources snapshot: iam-resources 1`] = `
{
  "permissions": {
    "WeeklyInactiveUserCheckRuleAllowEventRuleTestStackInactiveUserCheckerFunction4054795B2F7A268A": {
      "Properties": {
        "Action": "lambda:InvokeFunction",
        "FunctionName": {
          "Fn::GetAtt": [
            "InactiveUserCheckerFunctionD51D6C82",
            "Arn",
          ],
        },
        "Principal": "events.amazonaws.com",
        "SourceArn": {
          "Fn::GetAtt": [
            "WeeklyInactiveUserCheckRuleFB167B9C",
            "Arn",
          ],
        },
      },
      "Type": "AWS::Lambda::Permission",
    },
  },
  "policies": {
    "InactiveUserCheckerFunctionServiceRoleDefaultPolicy3B2516D9": {
      "Properties": {
        "PolicyDocument": {
          "Statement": [
            {
              "Action": [
                "secretsmanager:GetSecretValue",
                "secretsmanager:DescribeSecret",
              ],
              "Effect": "Allow",
              "Resource": {
                "Ref": "ApiSecretsED109BDA",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "InactiveUserCheckerFunctionServiceRoleDefaultPolicy3B2516D9",
        "Roles": [
          {
            "Ref": "InactiveUserCheckerFunctionServiceRoleF228211B",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
  },
  "roles": {
    "InactiveUserCheckerFunctionServiceRoleF228211B": {
      "Properties": {
        "AssumeRolePolicyDocument": {
          "Statement": [
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": [
          {
            "Fn::Join": [
              "",
              [
                "arn:",
                {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
              ],
            ],
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
  },
}
`;

exports[`CursorActiveUsersBotStack Infrastructure Tests Lambda Function should match Lambda function snapshot: lambda-functions 1`] = `
{
  "InactiveUserCheckerFunctionD51D6C82": {
    "DependsOn": [
      "InactiveUserCheckerFunctionServiceRoleDefaultPolicy3B2516D9",
      "InactiveUserCheckerFunctionServiceRoleF228211B",
    ],
    "Properties": {
      "Code": {
        "S3Bucket": "cdk-hnb659fds-assets-123456789012-us-east-1",
        "S3Key": "STATIC_ASSET_KEY.zip",
      },
      "Environment": {
        "Variables": {
          "ENABLE_CURSOR": "true",
          "ENABLE_GITHUB_COPILOT": "true",
          "ENABLE_NOTIFICATIONS": "true",
          "NOTIFY_AFTER_DAYS": "60",
          "REMOVE_AFTER_DAYS": "90",
          "SECRETS_ARN": {
            "Ref": "ApiSecretsED109BDA",
          },
        },
      },
      "Handler": "index.handler",
      "MemorySize": 512,
      "Role": {
        "Fn::GetAtt": [
          "InactiveUserCheckerFunctionServiceRoleF228211B",
          "Arn",
        ],
      },
      "Runtime": "nodejs22.x",
      "Timeout": 300,
    },
    "Type": "AWS::Lambda::Function",
  },
}
`;

exports[`CursorActiveUsersBotStack Infrastructure Tests Resource Counts should create the expected number of resources: resource-counts 1`] = `
{
  "AWS::Events::Rule": 1,
  "AWS::IAM::Policy": 1,
  "AWS::IAM::Role": 1,
  "AWS::Lambda::Function": 1,
  "AWS::Lambda::Permission": 1,
  "AWS::SecretsManager::Secret": 1,
}
`;

exports[`CursorActiveUsersBotStack Infrastructure Tests Secrets Manager should match Secrets Manager snapshot: secrets-manager 1`] = `
{
  "ApiSecretsED109BDA": {
    "DeletionPolicy": "Delete",
    "Properties": {
      "Description": "API keys for Cursor and Slack bots",
      "Name": "CursorActiveUserBotSecrets",
      "SecretString": "{"CURSOR_API_KEY":"YOUR_CURSOR_API_KEY_PLACEHOLDER","GITHUB_TOKEN":"YOUR_GITHUB_TOKEN_PLACEHOLDER","GITHUB_ORG":"YOUR_GITHUB_ORG_PLACEHOLDER","SLACK_BOT_TOKEN":"YOUR_SLACK_BOT_TOKEN_PLACEHOLDER","SLACK_USER_ID":"YOUR_SLACK_USER_ID_PLACEHOLDER","SLACK_SIGNING_SECRET":"YOUR_SLACK_SIGNING_SECRET_PLACEHOLDER"}",
    },
    "Type": "AWS::SecretsManager::Secret",
    "UpdateReplacePolicy": "Delete",
  },
}
`;
